### DISCUSSION TITLE
Adding custom filename extensions

### DISCUSSION THREAD

**Discussion by FJShen** [microsoft/vscode-cpptools#6868] [Q&A] [answered]
Is it possible to let the C/C++ extension also process files that ends with .cu and .cuh? I'm trying to code with CUDA, but this tool seems not to be processing those CUDA files, and I didn't find any high-quality extensions for CUDA-specific projects either. 

Any other workarounds are also welcome.

**Comment by bobbrow** [comment, marked as answer]
Take a look at the `Files: Associations` setting in VS Code.  If you tell VS Code to treat `*.cu` and `*.cuh` as `cpp`, then the C/C++ extension will start processing those files.  Do note that the extension does not currently understand CUDA-specific code, so you will see squiggles for those constructs.  You can work around some of those problems (e.g. certain keywords/attributes) by adding stuff like `"__global__="` in the `"defines"` section of your `c_cpp_properties.json` file.

**Reply by FJShen** [reply]
Thank you.
